plugins/code-climate/src/components/CodeClimateTable/CodeClimateTable.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { CodeClimateData } from '../../api';
+import clsx from 'clsx';
+import { InfoCard } from '@backstage/core-components';
+import { makeStyles, Typography } from '@material-ui/core';
+
+type CodeClimateTableProps = {
+  codeClimateData: CodeClimateData | null;
+};
+
+const letterStyle = {
+  color: 'white',
+  border: 0,
+  borderRadius: '3px',
+  fontSize: '40px',
+  padding: '5px 20px',
+};
+
+const fontSize = {
+  fontSize: '25px',
+};
+
+const letterColor = (letter: string) => {
+  if (letter === 'A') {
+    return '#45d298';
+  } else if (letter === 'B') {
+    return '#a5d86e';
+  } else if (letter === 'C') {
+    return '#f1ce0c';
+  } else if (letter === 'D') {
+    return '#f29141';
+  } else if (letter === 'F') {
+    return '#df5869';
+  }
+
+  return '#45d298';
+};
+
+const useStyles = makeStyles({
+  spaceAround: {
+    display: 'flex',
+    justifyContent: 'space-around',
+  },
+  spaceBetween: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  maintainabilityLetterColor: {
+    ...letterStyle,
+    backgroundColor: props =>
+      letterColor((props as any).maintainabilityLetter as string),
```

Comment:
here's an example of how you can set a props type on `makeStyles` to get type checking for the props https://github.com/backstage/backstage/blob/8651981680e76a8b1198075102d2dbc3ea9fbdc6/packages/core-components/src/components/Tabs/Tab.tsx#L40